arctic/store/version_store.py

```diff
@@ -500,16 +500,24 @@ def append(self, symbol, data, metadata=None, prune_previous_version=True, upser
         elif 'metadata' in previous_version:
             version['metadata'] = previous_version['metadata']
 
-        if handler and hasattr(handler, 'append'):
-            mongo_retry(handler.append)(self._arctic_lib, version, symbol, data,
-                                        previous_version, dirty_append=dirty_append, **kwargs)
+        if handler and hasattr(handler, 'append') and callable(handler.append):
+            handler.append(self._arctic_lib, version, symbol, data,
+                           previous_version, dirty_append=dirty_append, **kwargs)
         else:
             raise Exception("Append not implemented for handler %s" % handler)
 
         version['version'] = next_ver
 
         # Insert the new version into the version DB
-        mongo_retry(self._versions.insert_one)(version)
+        try:
```

Comment:
Repetition of this DuplicateKeyError handling block. Can you break it out into it's own def to avoid the duplication and reduce the risk of the multiple versions getting out of sync.

Reply:
thanks, I refactored adding _insert_version:
https://github.com/manahl/arctic/pull/487/commits/65f84c7134f714cf505f5d7208f18870e0761d1b